### DISCUSSION TITLE
@wagmi/core V1/V2: Track User Rejection Code

### DISCUSSION THREAD

**Discussion by Nilbristi** [wevm/wagmi#3428] [Q&A]
How to track the user rejection code while using writeContract.

In the below code, err function is getting fired for both user jection or regular failure case. I want to segregate the error by its code. Like if error code is 4001, it means User denied transaction signature.

I am getting the error code in inspect section but can't getting it in console.log. Check the attached screenshot

```
        await writeContract(config,  { request })
          .then((hash: WriteContractReturnType ) => {
            console.log('writeContract success', hash);
          },
          (err: WriteContractErrorType ) => {
            console.log('writeContract error', err);
          });
```

![image](https://github.com/wevm/wagmi/assets/282325/03aed6b1-7605-4d5a-952c-8d0eb5a60b48)

**Comment by cristobalbahe** [comment]
Would love this too 👌

**Comment by tmm** [comment]
The console error is getting thrown by the MM extension directly.
